executor.go

```diff
@@ -278,15 +279,26 @@ func executeFields(p ExecuteFieldsParams) *Result {
 		p.Fields = map[string][]*ast.Field{}
 	}
 
+	var wg sync.WaitGroup
+	wg.Add(len(p.Fields))
+
+	var resultsMutex sync.Mutex
 	finalResults := map[string]interface{}{}
 	for responseName, fieldASTs := range p.Fields {
-		resolved, state := resolveField(p.ExecutionContext, p.ParentType, p.Source, fieldASTs)
-		if state.hasNoFieldDefs {
-			continue
-		}
-		finalResults[responseName] = resolved
+		go func(responseName string, fieldASTs []*ast.Field) {
```

Comment:
This undoubtedly uses more resources due to the fact that is trying to do more at the same time. However, performance is immediately increased from parallelism when `GOMAXPROCS > 1`.

It especially makes sense when calling many long running procedures where the majority of the work is just waiting (i.e. RPCs).